.editorconfig

```diff
@@ -0,0 +1,14 @@
+# http://editorconfig.org
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
```

Comment:
I think this is not true for most of our scripts, as they don't contain final newline.

Reply:
This can be a source of all sorts of weirdnesses. According to the POSIX line definition, every line must have a newline at the end - final newline is no exception.
See 
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline
It's kind of especially important for the scripts since they are the most affected by portability issues. Realistically, even in 2019 we can still stumble upon a problematic interaction with improperly terminated files.
Good thing we can easily fix and enforce this with editor config.

Reply:
> It's kind of especially important for the scripts since they are the most affected by portability issues.

Note that most of our shell scripts depend on `bash` shell (as specified in the first line of each script), so they are already not POSIX-portable. `bash` handles scripts without newline in the end without issues, at least the scripts that we use.

I'm not saying that I'm against adding the newline in the end of each shell script, but I think it might make sense to discuss this separately beforehand, as part of https://github.com/timberio/vector/issues/1423 or as a separate issue. And if a decision is made on this, it should be enforced by a CI check.